osu.Game.Rulesets.Osu/Mods/OsuModZoomed.cs

```diff
@@ -0,0 +1,126 @@
+        public override ModType Type => ModType.Fun;
+        public override string Description => "Big brother is watching your cursor.";
+        public override double ScoreMultiplier => 1;
+
+        private const int last_zoom_combo = 200;
+        private const int zoom_every_combo_amount = 100;
+        private const int apply_zoom_duration = 1000;
+        private const double default_zoom = 1.8;
+        private const double zoom_with_combo_by = 0.1;
+
+        [SettingSource("Delay", "Delay in milliseconds for the view to catch up to the cursor")]
+        public BindableInt MovementDelay { get; } = new BindableInt
+        {
+            MinValue = 0,
+            MaxValue = 1000,
+            Precision = 100,
+        };
+
+        [SettingSource("Initial zoom", "The starting zoom level")]
+        public BindableDouble InitialZoom { get; } = new BindableDouble(default_zoom)
+        {
+            MinValue = 1.5,
+            MaxValue = 2,
+            Precision = 0.05
+        };
+
+        [SettingSource("Increase zoom with combo", "Whether zoom should increase as combo increases")]
+        public BindableBool ComboBasedZoom { get; } = new BindableBool(true);
+
+        private double currentZoom;
+
+        private readonly BindableInt currentCombo = new BindableInt();
+
+        private IFrameStableClock? gameplayClock;
+
+        public void ApplyToDrawableRuleset(DrawableRuleset<OsuHitObject> drawableRuleset)
+        {
+            currentZoom = InitialZoom.Value;
+            gameplayClock = drawableRuleset.FrameStableClock;
+        }
+
+        public void Update(Playfield playfield)
+        {
+            Debug.Assert(gameplayClock != null);
+
+            double currentScale = Interpolation.ValueAt(Math.Min(Math.Abs(gameplayClock.ElapsedFrameTime), apply_zoom_duration), playfield.Scale.X, currentZoom, 0, apply_zoom_duration, Easing.Out);
+
+            playfield.Scale = new Vector2((float)currentScale);
+
+            moveDrawablesFollowingCursor(playfield, gameplayClock);
+        }
+
+        private Vector2 getDrawablePositionForCursorPosition(Playfield playfield, Drawable drawable)
+        {
+            var position = playfield.Cursor.ActiveCursor.DrawPosition;
+            return Vector2.Clamp(playfield.OriginPosition - position, -playfield.OriginPosition, playfield.OriginPosition);
+        }
+
+        private void moveDrawablesFollowingCursor(Playfield playfield, IFrameStableClock gameplayClock)
+        {
+            var trackingPosition = getDrawablePositionForCursorPosition(playfield, playfield);
+
+            if (MovementDelay.Value == 0)
+                playfield.Position = trackingPosition;
+            else
+            {
+                playfield.Position = Interpolation.ValueAt(
+                    Math.Min(Math.Abs(gameplayClock.ElapsedFrameTime), MovementDelay.Value), playfield.Position, trackingPosition, 0, MovementDelay.Value, Easing.Out);
```

Comment:
Same here, I think you can use `DampContinuously`. It may even avoid requiring the special handling above to prevent div-by-zero.

Reply:
i changed it to that because otherwise it would be required to do these if statements checking for some edge cases where when using playback with `dampContinuously` would cause an infinity number thus making the game crash, so i was doing 2 if statements for checking if the numbers aren't infinity, with the interpolation alone this isn't necessary